packages/fxa-auth-server/lib/routes/subscriptions/paypal-notifications.ts

```diff
@@ -11,7 +13,77 @@ import { AuthLogger, AuthRequest } from '../../types';
 import { PayPalHandler } from './paypal';
 
 export class PayPalNotificationHandler extends PayPalHandler {
-  private async handleMerchPayment(message: IpnMerchPmtType) {}
+  /**
+   * Handle merchant payment notification from PayPal
+   * and update Stripe invoice according to the payment_status
+   *
+   * @param message
+   */
+  private async handleMerchPayment(message: IpnMerchPmtType) {
+    const invoice = await this.stripeHelper.getInvoice(message.invoice);
+    if (!invoice) {
+      this.log.error('handleMerchPayment', {
+        message: 'Invoice not found',
+        ipnMessage: message,
+      });
+      throw error.internalValidationError('handleMerchPayment', {
+        message: 'Invoice not found',
+      });
+    }
+
+    if (invoice.status == null || !['draft', 'open'].includes(invoice.status)) {
+      this.log.error('handleMerchPayment', {
+        message: 'Invoice not in draft or open status',
+        ipnMessage: message,
+      });
+      throw error.internalValidationError('handleMerchPayment', {
```

Comment:
When we run the payment during checkout and it succeeds, we immediately pay the invoice and its in a paid status. Which means this event will throw an error. The easy way to know we can ignore this transaction would be to see if the invoice metadata has a transaction id that matches the transaction id on this message, if so it was paid already and we can stop.